### PR TITLE
fix: changed inputprops to fontweight and fontsize to avoid overlap with html native

### PR DESCRIPTION
**Pull request by charlielizzy**
added fontWeight and fontSize props to InputProps to avoid naming of weight and size overlapping with existing props

### CODE REVIEW

File: src/components/atoms/InputText/InputText.tsx [hunk 1]
```diff
@@ -57,7 +57,7 @@ const Input = styled.input<InputProps & { theme: AppTheme }>`
   padding-left: ${({ startIcon }) => !!startIcon && '60px'};
   padding-right: ${({ endIcon }) => !!endIcon && '60px'};
   font-size: ${({ theme, size = 'body' }) => theme.typography.text.sizes[size]};
```
Comment (OlenaKashuba) [inline src/components/atoms/InputText/InputText.tsx:59]: you need to use {theme, fontSize} - new prop you renamed

**Review by poteirard — APPROVED** [review]

File: src/components/types.ts [hunk 2]
```diff
@@ -32,7 +32,8 @@ export interface InputProps extends InputStatus, InputHTMLAttributes<HTMLInputEl
    */
   endIcon?: ReactNode;
 
-  weight?: number;
+  fontWeight?: number;
+  fontSize?: string;
```
Comment (poteirard) [inline src/components/types.ts:36]: Do we have full freedom of these strings and number? Shall we use a set of options instead?

**Review by poteirard — CHANGES_REQUESTED** [review]

File: src/components/types.ts [hunk 3]
```diff
@@ -32,7 +32,8 @@ export interface InputProps extends InputStatus, InputHTMLAttributes<HTMLInputEl
    */
   endIcon?: ReactNode;
 
-  weight?: number;
+  fontWeight?: number;
```
Comment (moshie) [inline src/components/types.ts:35]: Shouldn't this be a string given how it's used here:

```
${({ theme, fontWeight = 'regular' }) => theme.typography.weights[fontWeight]};
```

Comment (OlenaKashuba) [reply src/components/types.ts:35]: no, fontWeight is a number https://github.com/zopaUK/react-components/blob/master/src/components/styles/Theme/ThemeProvider.tsx#L241

Comment (OlenaKashuba) [reply src/components/types.ts:35]: https://github.com/zopaUK/react-components/blob/master/src/constants/typography.ts#L32

File: src/components/types.ts (hunk 2 above)
Comment (charlielizzy) [reply src/components/types.ts]: good shout - i've added in custom types for FontWeights and FontSizes now :)

File: src/components/types.ts (hunk 3 above)
Comment (moshie) [reply src/components/types.ts]: InputProps is being used as the interface for this:

https://github.com/zopaUK/react-components/pull/860/files#diff-7fba51a07023554f00291cb6d526da934112691ee58c2ed0e1f37ffe3e8bd60eR50

and the default here: `fontWeight = 'regular'` is a string which makes no sense

Comment (charlielizzy) [reply src/components/types.ts]: i've made changes to this now :)

File: src/components/types.ts [hunk 4]
```diff
@@ -32,7 +32,9 @@ export interface InputProps extends InputStatus, InputHTMLAttributes<HTMLInputEl
    */
   endIcon?: ReactNode;
 
-  weight?: number;
+  fontWeight?: keyof AppTheme['typography']['weights'];
```
Comment (moshie) [inline src/components/types.ts:35]: The type of `keyof AppTheme['typography']['weights'];` this is an object not a string or a number?

https://github.com/zopaUK/react-components/blob/master/src/components/styles/Theme/ThemeProvider.tsx#L240-L245

**Review by moshie — CHANGES_REQUESTED** [review]

**Review by moshie — APPROVED** [review]

**Pull request closed without merging**